docs/debug.md

```diff
@@ -103,6 +103,57 @@ While pulling, the image manifest, config, and layers without zTOCs' are fetched
 
 Below are a list of common error paths that may occur in this phase:
 
+### Determining how an image was pulled
+
+The SOCI snapshotter can pull an image one of 4 modes:
+1) Lazily using an explicitly provided SOCI index digest
+2) Lazily using a SOCI index manifest v2 via SOCI-enabled images
+3) Lazily using a SOCI index manifest v1 via the Referrers API
+4) Ahead of time without a SOCI index
+
+Determining which of these mode was used is often a good first step for debugging image pull issues.
+
+When SOCI starts to pull an image, it will emit a set of debug logs to indicate which modes were tried and which was ultimately used.
+
+#### Explicit index digest
+If SOCI finds an explicit index digest, it will use it and log the following message:
+```
+using provided soci index digest
+```
+
+#### SOCI index manifest v2 via SOCI-enabled images
+If an explicit index digest was not provided. SOCI will try to use a SOCI-enabled image.
+
+if `pull_modes.soci_v2.enaled` is false, the snapshotter will log
+```
+soci v2 is disabled
+```
+
+Otherwise, if it find that the image is SOCI-enabled, it will log:
+```
+using soci v2 index annotation
+```
+
+#### Lazily using a SOCI index manifest v1 via the Referrers API
+If the image is not SOCI-enabled, the snapshotter will try to use the referrers API.
+
+if `pull_modes.soci_v1.enaled` is false (it is false by default), the snapshotter will log
+```
+soci v1 is disabled
+```
+
+Otherwise, if it find that the image is SOCI-enabled, it will log:
```

Comment:
Maybe this should say the image is SOCI-enabled through index lookup using the referrers API to distinguish it a bit more from the above write-up on v2.

Reply:
Yeah, this was a bad copy-paste. Will fix.